apisix/core/config_etcd.lua

```diff
@@ -157,45 +369,71 @@ local function flush_watching_streams(self)
 end
 
 
-local function http_waitdir(etcd_cli, key, modified_index, timeout)
-    local opts = {}
-    opts.start_revision = modified_index
-    opts.timeout = timeout
-    opts.need_cancel = true
-    local res_func, func_err, http_cli = etcd_cli:watchdir(key, opts)
-    if not res_func then
-        return nil, func_err
+local function http_waitdir(self, etcd_cli, key, modified_index, timeout)
+    if not watch_ctx.idx[key] then
+        watch_ctx.idx[key] = 1
```

Comment:
could we store the min_idx and assgin here?

Reply:
It's used to init the idx per resource. The min_idx is the global status for all resources. What do you mean?

Reply:
Each resource must start from 1, because it needs to consume the earliest event!

Reply:
But the resource before the mini_idx have been cleaned?
```
            for i = 1, min_idx - 1 do
                watch_ctx.res[i] = false
            end
```

Reply:
It will skip the false items. When it starts to match, the queue may be not cleaned.

Reply:
so store the min_idx, and new resource could start from min_idx, because the event before min_idx is false, right ?

Reply:
1. min_idx maybe 0, which is an internal state of the main watcher and comes from the summary of consumer idx of all resources
2. after the first watch iteration, the consume idx of the resource will be kept in sync with the global min_idx.
3. the sync gap is small, now the max gap is 100, so the overhead should be ignored

Reply:
ok